lib: updatehub: Add missing include for FLASH_AREA

nandojve opened the pull request:
The conversion from DT_FLASH_AREA to FLASH_AREA macros don't add the
storage flash_map.h include file.

Fixes: #25332

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

nandojve committed:
lib: updatehub: Add missing include for FLASH_AREA

The conversion from DT_FLASH_AREA to FLASH_AREA macros don't add the
storage flash_map.h include file.

Fixes: #25332

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

otavio commented:
I'd like to understand how this breakage has not been spotted by the CI system? This is something we should verify so we avoid these build failures in the future.

otavio commented:
It seems it was caused by #25056, which @galak merged, however it seems it was not tested on CI.

otavio commented:
Looking on the CI result, it runned all CI checks so our sample is not enabled for it.

@nandojve can we enable it to be tested on CI?

nandojve committed:
lib: updatehub: Fix compile warning

Based on the current platform a warning can raise becase of missing
string.h include file.

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

nandojve committed:
samples: net: updatehub: Add sample.yaml file

Add missing sample.yaml file to enable CI build tests. Local tests can
be executed executing sanitycheck:

./scripts/sanitycheck -l --testcase-root samples/net/updatehub

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

nandojve commented:
Two new commits:
1- Fix compile warning
2- Add missing sample.yaml to enable CI build sample/net/updatehub
